apps/api/src/app/events/usecases/create-notification-jobs/create-notification-jobs.usecase.ts

```diff
@@ -142,4 +140,51 @@ export class CreateNotificationJobs {
 
     return steps;
   }
+
+  private calculateExpireAt(command: CreateNotificationJobsCommand) {
+    const delayedSteps = command.template.steps.filter(
+      (step) => step.template?.type === StepTypeEnum.DIGEST || step.template?.type === StepTypeEnum.DELAY
+    );
+
+    const delay = delayedSteps
+      .map((step) => this.calculateDelayAmount(step, command.payload, command.overrides))
+      .reduce((partialSum, a) => partialSum + a, 0);
+
+    return addMilliseconds(Date.now(), delay);
+  }
```

Comment:
@ainouzgali where does all of this functionality tested? would suggest extracting this to a class with public methods, and tests those public methods with unit tests, there is a lot of code here and would be great to test for it :)

Reply:
extracted it, this was duplicated from the logic in add-delay-job